lib/ansible/galaxy/collection.py

```diff
@@ -872,16 +873,21 @@ def _get_galaxy_yml(b_galaxy_yml_path):
         raise AnsibleError("Failed to parse the galaxy.yml at '%s' with the following error:\n%s"
                            % (to_native(b_galaxy_yml_path), to_native(err)))
 
-    set_keys = set(galaxy_yml.keys())
-    missing_keys = mandatory_keys.difference(set_keys)
-    if missing_keys:
-        raise AnsibleError("The collection galaxy.yml at '%s' is missing the following mandatory keys: %s"
-                           % (to_native(b_galaxy_yml_path), ", ".join(sorted(missing_keys))))
+    # Sanity test on the contents of the galaxy.yml file
+    if not isinstance(galaxy_yml, MutableMapping):
```

Comment:
Should this also take into account whether list is running into a galaxy.yml like this and to be more resilient?

Reply:
I'm not sure what you mean. 😕

Reply:
Sorry for losing track of this one. I was concerned that this code path would traceback for `list` instead of just carrying on to the next collection like missing keys or an empty file would do. Since a collection can contain anything in galaxy.yml or MANIFEST.json and still be functional for playbooks to use, `--list` should probably ignore non-dictionary galaxy.yml files in addition to not requiring mandatory keys.